### ISSUE TITLE
split_report is O(n^2) in the full dataset; max_samples does not cover the geometric metrics

### ISSUE THREAD

**Issue by gxlarson** [gxlarson/splytters#65]
`split_report` is O(n²) in the **full** dataset, so it is unusable on datasets the splitters themselves handle fine. `max_samples` does not cover the geometric metrics.

Found while running `per_class_split(distance_adversarial_split, ...)` on 399,999 CLIP ViT-B/32 embeddings (400k × 512). The split itself took **0.4 s**. `split_report` on the result never returned.

## Cause

`split_report` subsamples to `max_samples` only for the *distribution* metrics (`report.py:225-226`), but at `report.py:215` it calls:

```python
report.update(compute_split_similarity(X, train_indices, test_indices, metric))
```

with the **full, unsubsampled** `X`. `compute_split_similarity` (`utils.py:304`) then does two quadratic things:

- `utils.py:336` — `cdist(test_X, train_X, metric=metric)`, i.e. `|test| × |train|`
- `utils.py:343` — `cdist(X, X, metric=metric)`, a full `n × n` matrix, allocated **only** to take a median for the `coverage` statistic

At n = 399,999 that second line alone asks for 399999² × 8 B = **1.28 TB**. There is already a `TODO` acknowledging this at `utils.py:341`.

## Repro

```python
import numpy as np, splytters as s
X = np.random.default_rng(0).normal(size=(50_000, 64))
tr, te = s.random_split(X, 0.8, 42)
s.split_report(X, tr, te, max_samples=2000)   # 20 GB alloc; thrashes or OOMs
```

Scaling measured on 0.2.1 (`max_samples=2000` throughout, so the growth is entirely the uncapped part):

| n | `split_report` | `cdist(X, X)` float64 |
|---|---|---|
| 1,000 | 2.1 s | 0.01 GB |
| 2,000 | 4.7 s | 0.03 GB |
| 4,000 | 6.7 s | 0.13 GB |
| 8,000 | 8.9 s | 0.51 GB |
| 16,000 | 16.5 s | 2.05 GB |
| 399,999 | — | 1,280 GB |

## Why it matters

`distance_adversarial_split` is O(n·d + n log n) and splits 400k points in under a second, but there is then no supported way to characterize the split it produced. The workaround is to subsample the *indices* after splitting and rebuild a small `X` — but that is easy to get wrong (subsampling the data before splitting measures a different split's geometry).

## Suggested fix

1. Honor `max_samples` in the geometric metrics too — either subsample inside `compute_split_similarity`, or subsample `X`/indices in `split_report` before the `report.py:215` call.
2. Estimate `median_dist` from a random sample of pairs instead of materializing `cdist(X, X)`; the statistic is a median, so a few million sampled pairs are plenty.
3. Use `sklearn.neighbors.NearestNeighbors` for `min_distances` to get O(n log n) instead of `|test| × |train|`.
4. Document that `max_samples` caps *all* pairwise work, once it does.

Happy to send a PR if the approach looks right.

Found on `main` @ `ed2f0e7` (= PyPI 0.2.1).
